litentry/core/ruleset-build/src/discord.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{CheckJoinDiscordResponse, Error, Result};
+
+use itc_rest_client::{
+	http_client::{DefaultSend, HttpClient},
+	rest_client::RestClient,
+	RestGet,
+};
+use std::{format, str, string::String, time::Duration, vec};
+use url::Url;
+
+use itp_types::AccountId;
+
+use litentry_primitives::{Identity, ParameterString};
+
+const DISCORD_BASE_URL: &str = "https://47.57.13.126:8080/";
+const TIMEOUT: Duration = Duration::from_secs(3u64);
+
+pub fn ruleset2_verification(
+	_who: AccountId,
+	_identity: Identity,
+	guild_id: ParameterString,
+	user_id: ParameterString,
+) -> Result<()> {
+	let base_url = Url::parse(DISCORD_BASE_URL).unwrap();
+	let http_client = HttpClient::new(DefaultSend {}, true, Some(TIMEOUT), None, None);
+	let mut client = RestClient::new(http_client, base_url);
+
+	let path = format!(
+		"/discord/joined?guildid={:?}&userid={:?}",
+		guild_id.into_inner(),
+		user_id.into_inner()
+	);
+	let query = vec![];
+
+	let response: CheckJoinDiscordResponse = client
+		.get_with::<String, CheckJoinDiscordResponse>(path, query.as_slice())
+		.map_err(|e| Error::Ruleset1Error(format!("{:?}", e)))?;
```

Comment:
I didn't test it, is this enough? Do we get errors when the user didn't join the guild?
I see there's `has_errors` field but it seems unused?

Reply:
I think `has_errors` should be used when generating VC.